pkg/doc/signature/suite/jsonwebsignature2020/public_key_verifier.go

```diff
@@ -7,186 +7,22 @@ SPDX-License-Identifier: Apache-2.0
 package jsonwebsignature2020
 
 import (
-	"crypto"
-	"crypto/ecdsa"
-	"crypto/ed25519"
-	"crypto/elliptic"
-	"crypto/rsa"
-	"crypto/x509"
-	"errors"
-	"fmt"
-	"math/big"
-
-	"github.com/btcsuite/btcd/btcec"
-
-	sigverifier "github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
 )
 
-// PublicKeyVerifier verifies a Ed25519 / EC (P-256, P-384, P-521, secp256k1) / RSA signature
-// taking public key bytes and / or JSON Web Key as input.
+// NewPublicKeyVerifier creates a signature verifier that verifies a Ed25519 / EC (P-256, P-384, P-521, secp256k1) / RSA
+// signature taking public key bytes and / or JSON Web Key as input.
 // The list of Supported JWS algorithms of JsonWebSignature2020 is defined here:
 // https://github.com/transmute-industries/lds-jws2020#supported-jws-algs
-// NOTE: this verifier is present for backward compatibility reasons and can be removed later.
-// Please use CryptoVerifier or your own verifier.
-type PublicKeyVerifier struct {
-}
-
-// Verify will verify a signature.
-func (v *PublicKeyVerifier) Verify(pubKey *sigverifier.PublicKey, doc, signature []byte) error {
-	// A presence of JSON Web Key is mandatory (due to JwsVerificationKey2020 type).
-	if pubKey.JWK == nil {
-		return ErrJWKNotPresent
-	}
-
-	if pubKey.Type != jwkType {
-		return ErrTypeNotJwsVerificationKey2020
-	}
-
-	switch pubKey.JWK.Kty {
-	case "EC":
-		return v.verifyEllipticCurve(pubKey, signature, doc)
-	case "OKP":
-		return v.verifyEdDSA(pubKey, signature, doc)
-	case "RSA":
-		return v.verifyRSA(pubKey, signature, doc)
-	default:
-		return fmt.Errorf("unsupported key type: %s", pubKey.JWK.Kty)
-	}
-}
-
-func (v *PublicKeyVerifier) verifyEllipticCurve(pubKey *sigverifier.PublicKey, signature, msg []byte) error {
-	ec := parseEllipticCurve(pubKey.JWK.Crv)
-	if ec == nil {
-		return fmt.Errorf("ecdsa: unsupported elliptic curve '%s'", pubKey.JWK.Crv)
-	}
-
-	pubKeyBytes := pubKey.Value
-
-	x, y := elliptic.Unmarshal(ec.Curve, pubKeyBytes)
-	if x == nil {
-		return errors.New("ecdsa: invalid public key")
-	}
-
-	ecdsaPubKey := &ecdsa.PublicKey{
-		Curve: ec.Curve,
-		X:     x,
-		Y:     y,
-	}
-
-	if len(signature) != 2*ec.KeySize {
-		return errors.New("ecdsa: invalid signature size")
-	}
-
-	hasher := crypto.SHA256.New()
-
-	_, err := hasher.Write(msg)
-	if err != nil {
-		return errors.New("ecdsa: hash error")
-	}
-
-	hash := hasher.Sum(nil)
-
-	r := big.NewInt(0).SetBytes(signature[:ec.KeySize])
-	s := big.NewInt(0).SetBytes(signature[ec.KeySize:])
-
-	verified := ecdsa.Verify(ecdsaPubKey, hash, r, s)
-	if !verified {
-		return errors.New("ecdsa: invalid signature")
-	}
-
-	return nil
-}
-
-type ellipticCurve struct {
-	Curve   elliptic.Curve
-	KeySize int
+func NewPublicKeyVerifier() *verifier.PublicKeyVerifier {
```

Comment:
This is how a signature verifier for the composite signature suite is created.

Reply:
We re-use `Ed25519SignatureVerifier` and `ECDSASecp256k1SignatureVerifier` here.